apidef/oas/Taskfile.yml

```diff
@@ -0,0 +1,26 @@
+---
+version: "3"
+
+tasks:
+  default:
+    desc: "Run everything"
+    cmds:
+      - task: test
+      - task: lint
+
+  test:
+    desc: "Run tests"
+    cmds:
+      - go fmt ./...
+      - go test -count=1 ./...
+
+  # lint target is run from CI
+  lint:
+    desc: "Ensure linter pass"
+    cmds:
+      - schema-gen extract -o - | schema-gen lint --summary -i -
+
+  deps:
```

Comment:
what is upgrade policy for schema-gen in oas and config Taskfile? Should it be manual or automated by including deps as part of lint task?

Reply:
CI:

- It's automated by CI (on every tooling change, weekly).
- https://github.com/TykTechnologies/tyk-github-actions/blob/main/docker/tools/latest/Dockerfile

Development:

For manual updates of tooling we don't have a policy, but if you rely on it then you should keep your local versions up to date. Docker is also an option, something like this should use the tooling that CI uses:

```
docker run --platform linux/amd64 --rm -v $PWD:$PWD -w $PWD tykio/ci-tools:latest task
```